hs_access_control/models/community.py

```diff
@@ -8,6 +10,12 @@
 ###################################
 # Communities of groups
 ###################################
+def get_upload_path(instance, filename):
+    file_base, file_ext = os.path.splitext(filename)
+    unique_id = uuid4().hex
+    name = instance.name
```

Comment:
instead of taking instance as a parameter to this method, you could take name.  Then create wrappers that parse the different types and call it.  That would allow you to avoid circular imports and get code reuse.

```
def get_upload_path(name, filename):
    file_base, file_ext = os.path.splitext(filename)
    unique_id = uuid4().hex
    return f'group/{file_base}_{name}_{unique_id}{file_ext}'

def get_upload_path_community(instance, filename):
   return get_upload_path(instance.name, filename)

def get_upload_path_group(instance, filename):
   return get_upload_path(instance.group.name, filename)

def get_upload_path_user(instance, filename):
   return get_upload_path(instance.user.username, filename)

```

Reply:
@sblack-usu No, it is not creating a different folder for each file. As currently implemented, all user profile uploaded pictures are placed in ```profile``` folder, all group uploaded pictures are placed in ```group``` folder, and all community uploaded pictures are placed in ```community``` folder. I am keeping the same folder organization. Let me know if this is still not clear.

Reply:
I'm good with this if the util method organization is addressed.